tests/cpu_bugs/xen_mitigations.pm

```diff
@@ -0,0 +1,349 @@
+            }
+        },
+        "eager-fpu=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB EAGER_FPU MD_CLEAR']},
+                unexpection => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB MD_CLEAR']}
+            }
+        },
+        "l1d-flush=off" => {
+            "default" => {
+                expection   => {''},
+                unexpection => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+            }
+        },
+        "l1d-flush=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+                unexpection => {''},
+            }
+        },
+    },
+  };
+sub do_check {
+    my $secnario = shift;
+    my $foo      = $secnario->{'default'};
+    if ($foo->{'expection'}) {
+        while (my ($cmd, $lines) = each %{$foo->{'expection'}}) {
+            foreach my $l (@{$lines}) {
+                if ($l ne "") {
+                    assert_script_run("$cmd | grep \"$l\"");
+                } else {
+                    print "This expection is empty string, skip";
+                }
+
+            }
+        }
+    }
+    if ($foo->{'unexpection'}) {
+        while (my ($cmd, $lines) = each %{$foo->{'unexpection'}}) {
+            foreach my $l (@{$lines}) {
+                if ($l ne "") {
+                    my $ret = script_run("$cmd | grep \"$l\"");
+                    if ($ret eq 0) {
+                        die "found a unexpection string.";
+                    }
+                } else {
+                    #Debug what output be report.
+                    assert_script_run("xl dmesg | grep -A 10 \"Speculative\"");
+                    print "This unexpection is empty string, skip";
+
+                }
+
+            }
+        }
+    }
+}
+
+sub do_test {
+    my ($self, $hash) = @_;
+    #xen parameter be store into arg
+    #
+    while (my ($arg, $dict) = each %$hash) {
+        while (my ($key, $value) = each %$dict) {
+            my $parameter = $arg . '=' . $key;
+            bootloader_setup::add_grub_xen_cmdline_settings($parameter);
+            bootloader_setup::grub_mkconfig();
+            Mitigation::reboot_and_wait($self, 150);
+            #$value include the check rules of current $parameter.
+            my $ret = do_check($value);
+            if ($ret ne 0) {
+                record_info('ERROR', "$parameter test is failed.", result => 'fail');
+                die;
+            }
+            record_info('INFO', "$parameter test is finished.");
+            bootloader_setup::remove_grub_xen_cmdline_settings($parameter);
+            bootloader_setup::grub_mkconfig();
+        }
+
+    }
+}
+
+
+sub run {
+    my $self = @_;
+    select_console 'root-console';
```

Comment:
have you consider using serial terminal for this tests ? 
```suggestion
    $self->select_serial_terminal;
```

Reply:
sorry, I don't know this function. My "root-console" should be "ssh_console_for_ipmi_backend", your select_serial_terminal is this ?

Reply:
@pevik  do you know if we can use serial terminal for xen ?

Reply:
@asmorodskyi @wjn740 actually it'd be nice to test `select_serial_terminal()` with XEN and extend it for XEN if modifications are needed.

Reply:
BTW there is ticket for adding XEN support to svirt serial https://progress.opensuse.org/issues/55985
I guess some testing of libvirt configuration is needed (IMHO not that difficult, but I haven't managed to find time to do it). If you're interested to implement it, I'll be happy to help you.

Reply:
@pevik sorry, I think I have no extra time for the ticket of openQA.

Reply:
@wjn740 no problem.
@Martchus @foursixnine @okurz anybody interested in implementing https://progress.opensuse.org/issues/55985 ?